cmd/agent/subcommands/run/command.go

```diff
@@ -354,6 +358,9 @@ func getSharedFxOption() fx.Option {
 		}),
 		ndmtmp.Bundle,
 		netflow.Bundle,
+		fx.Provide(func(demultiplexer demultiplexer.Component) optional.Option[collector.Collector] {
+			return optional.NewOption(common.LoadCollector(demultiplexer))
```

Comment:
Does your code need a full collector or only the method `AddEventReceiver`? In the later case you can create a new interface with only `AddEventReceiver`  and provides a noop implementation when needed. The benefit is when collector component will be created and used, function doesn't necessary have to handle the optionality for the component. What do you think?

Reply:
>  do think a better design would be to have a "noop" version of the collector that is provided by the main instead of an Optional but I think that calls for another PR.

In that case, can you add a comment?